osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/SecretManagementService.java

```diff
@@ -1,181 +1,317 @@
+     * the public key from an asymmetrical key pair for the platform, which is
+     * available to external organizations.<br>
+     * Inside the DLMS protocol adapter keys are encrypted with a faster
+     * symmetrical encryption using a secret key that is not supposed to be
+     * known outside this protocol adapter.
+     *
+     * @param externallyEncryptedKey
+     *         key encrypted with the externally known public key for OSGP
+     *
+     * @return the key encrypted with the symmetrical secret key used only
+     *         inside the DLMS protocol adapter, or an empty byte array if
+     *         {@code externallyEncryptedKey == null}
+     *
+     * @throws FunctionalException
+     *         in case of a encryption/decryption errors while handling the
+     *         key
+     */
     @Override
-    public byte[] reEncryptKey(byte[] externallyEncryptedKey, SecurityKeyType keyType) throws FunctionalException {
-        throw new NotImplementedException();
-    }
+    public byte[] reEncryptKey(final byte[] externallyEncryptedKey/*, final SecurityKeyType keyType*/)
+            throws FunctionalException {
 
-    @Override
-    public byte[] decryptKey(byte[] encryptedKey, SecurityKeyType keyType) throws ProtocolAdapterException {
-        throw new NotImplementedException();
-    }
+        if (externallyEncryptedKey == null) {
+            return new byte[0];
+        }
 
-    @Override
-    public byte[] encryptKey(byte[] plainKey, SecurityKeyType keyType) throws ProtocolAdapterException {
-        throw new NotImplementedException();
-    }
+        final byte[] key = this.rsaDecrypt(externallyEncryptedKey);//, keyType);
+        return this.aesEncryptKey(key);//, keyType);
 
-    @Override
-    public byte[] getDlmsMasterKey(String deviceIdentification) {
-        throw new NotImplementedException();
     }
 
     @Override
-    public byte[] getDlmsAuthenticationKey(String deviceIdentification) {
+    public byte[] rsaDecrypt(final byte[] externallyEncryptedKey) //, final SecurityKeyType keyType)
+            throws FunctionalException {
 
         try {
-            GetSecretsRequest request = getSoapRequestForKey(deviceIdentification,
-                    SecretType.E_METER_AUTHENTICATION_KEY);
-            GetSecretsResponse response = secretManagementClient.getSecretsRequest(request);
-            Optional<TypedSecret> optionalTypedSecret = getTypedSecretFromSoapResponse(response,
-                    SecretType.E_METER_AUTHENTICATION_KEY);
+            return this.rsaEncryptionService.decrypt(externallyEncryptedKey);
+        } catch (final Exception e) {
+            LOGGER.error("Unexpected exception during decryption", e);
+            throw new FunctionalException(FunctionalExceptionType.DECRYPTION_EXCEPTION, ComponentType.PROTOCOL_DLMS, e);
+            // new EncrypterException(
+            //         String.format("Unexpected exception during decryption of key.", keyType)));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)